### PR TITLE
Issue 1 launguage order backend queries

### LINKED ISSUE #5826: Languages order preference
Some screens show labels (and depictions) of Wikidata items. If the Wikidata item does not have a label, the app picks a label in any other language.

Often, the app shows a label in a language I can't read (Turkish in the screenshot below) even though the Wikidata item has a label in a language I can read (such as Spanish):

It would be great if the settings let me choose not just one language (such as French in the screenshot below), but several languages with an order, for instance French,Spanish,Italian,English.

The UI could look like the Android OS language setting, but any better UI is welcome:

The setting would modify the SPARQL queries sent by the app (don't worry if you don't know SPARQL, I can help with that part).

https://github.com/commons-app/apps-android-commons/blob/c3cd30ce5c88ad3abfbd77b59e2460585a3ff720/app/src/main/resources/queries/query_for_item.rq#L20

### PR DESCRIPTION
**Pull request by Adeeth101**
**Languages order preference UI and functionality enhancement.**

Fixes #5826

What changes did you make and why?

Gradle Files: 
Some fo the gradle code was outdated and didn't work on our latest emulators. I had to remove code that was no longer supported.

Preferences.XML: 
Added a new preference category in the preferences UI for secondary language picking
Prefs.java was also changed to hold a serialised string of secondary language codes

Strings.XML: 
Added new strings that represent additions in UI

SettingsFragments: 
- Had to accommodate new preference in OnCreate
- Made a new dialog called prepareSecondaryLanguageDialog(). 
- Made helper methods for that dialog that use a new language adapter SavedLanguagesAdapter
- modified saving and getting key value methods to accommodate the new category in Prefs

NearbyController and NearbyPlaces:
Injected the KV store into the controller and passed the serialised string as a new argument to the json API client in NearbyPlaces

OkHttpJsonApiClient:
changed the getPlaces method to make use of the new parameter (the serialised language code)
made stringbuilders to automatically generate further SparQL queries based on the serialised code list
(Also modified the SparQL file itself to add placeholders for the additional strings created by the method)

**Tests performed (required)**
Apart from extensive manual software testing, the following unit tests were created:
StringBuilderTest.java
SettingsFragmentSecondaryLanguageTests.kt

Tested {build variant, e.g. ProdDebug} on {name of device or emulator} with API level {API level}.

**Screenshots (for UI changes only)**
image
image

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Thanks a lot!

Sorry I am outside and can't test right now.

Would you mind describing how I should do to switch German (de) and Japanese (ja)?

Any specific reason to split primary and secondary, rather than having just one list of languages? It is fine but just curious what your thoughts are om this.

**Comment by Adeeth101** [comment]
Ill work on name changes and the change in the res file is unintentional sorry about that...

File: app/src/main/res/values/arrays.xml [hunk 1]
```diff
@@ -25,6 +25,17 @@
         <item>@string/exif_tag_name_serialNumbers</item>
         <item>@string/exif_tag_name_software</item>
     </array>
+
+    <array name="default_languages">
```
[the remark this answers is not in the document]
Comment (Adeeth101) [reply app/src/main/res/values/arrays.xml:29]: That wass a mistake, apologies.

File: app/src/main/res/layout/dialog_select_secondary_language.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+<androidx.constraintlayout.widget.ConstraintLayout
```
[the remark this answers is not in the document]
Comment (Adeeth101) [reply app/src/main/res/layout/dialog_select_secondary_language.xml:1]: I could look into that but would that be something that is absolutely needed for the PR?

File: app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java [hunk 3]
```diff
@@ -267,6 +290,170 @@ private void prepareTheme() {
...
+        String[] languageArray = languageCodes.split(",\\s*");  // Split by comma and optional space
+        return new ArrayList<>(Arrays.asList(languageArray));  // Convert array to ArrayList and return
+    }
+
+    /**
+     * Prepare and Show language selection dialog box
+     * Disable default/already selected language from dialog box
+     * Saves values chosen by user to shared preferences as a serialised string.
+     */
+    private void prepareSecondaryLanguageDialog() {
```
[the remark this answers is not in the document]
Comment (Adeeth101) [reply app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java:334]: Ill do that for the rest of the plural cases.

File: app/src/main/res/layout/dialog_select_secondary_language.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicolas-raoul) [reply app/src/main/res/layout/dialog_select_secondary_language.xml:1]: Not absolutely needed, please just create a new enhancement issue about it if you don't have time to do it. Thanks! :-)

**Comment by nicolas-raoul** [comment]
Also, please rebase or merge from master. Thanks! :-)

**Comment by Adeeth101** [comment]
@nicolas-raoul Ive squashed merge conflicts and resolved above points.

**Comment by nicolas-raoul** [comment]
I am so sorry for leaving it for so long, would you mind this other conflict? Thanks a lot!

**Comment by Adeeth101** [comment]
Ill look into it

**Comment by Adeeth101** [comment]
@nicolas-raoul Sorry again for being late. Ill fix it today.

**Comment by Adith101** [comment]
Hi @nicolas-raoul this is (@Adeeth101) im on my alt account because I'm on my phone. 

A lot of the java classes I edited are now deleted and replaced kotlin versions. Not a huge deal but it does mean I basically cant auto merge and it will take some time to redo my work in kotlin. Give me another day please and thanks for the patience.

**Comment by Adeeth101** [comment]
I made a new MR, which should now be good for the updated Kotlin version of the app.

**Pull request closed without merging**